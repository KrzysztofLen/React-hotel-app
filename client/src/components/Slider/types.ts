export type Props = {
  readonly images: Array<string>;
};
