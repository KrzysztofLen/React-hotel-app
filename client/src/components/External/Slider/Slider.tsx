import React, {Component} from "react";
import SliderItem from "./SliderItem/SliderItem";

interface IProps {
    images: Array<string> | undefined
}

interface IState {
    currentIndex: number
}

class Slider extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);

        this.state = {
            currentIndex: 1
        }
    };

    onCloseModal(event: Event) {
        if (event != undefined) {
            event.preventDefault();
        }
    }

    onPrev = (event: Event) => {
        if (event != undefined) {
            event.preventDefault();
        }

        this.setState(prevState => ({
            currentIndex: prevState.currentIndex - 1
        }));
    }

    onNext = (event: Event) => {
        if (event != undefined) {
            event.preventDefault();
        }

        this.setState(prevState => ({
            currentIndex: prevState.currentIndex + 1
        }));
    }

    render() {
        if (this.props.images == null) {
            return false;
        }

        return (
            <SliderItem hasNext={this.state.currentIndex + 1 < this.props.images.length}
                        hasPrev={this.state.currentIndex > 0}
                        onNext={this.onNext}
                        onPrev={this.onPrev}
                        onCloseModal={this.onCloseModal}
                        src={this.props.images[this.state.currentIndex]}/>
        )
    }
}

export default Slider;