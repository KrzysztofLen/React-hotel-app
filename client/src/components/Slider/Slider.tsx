import React, { useState } from 'react';
import SliderItem from './SliderItem/SliderItem';
import { Props } from './types';

const Slider = (props: Props) => {
  const [currentIndex, setCurrentIndex] = useState<number>(1);

  const onCloseModal = (event: Event): void => {
    if (event != undefined) {
      event.preventDefault();
    }
  };

  const onPrev = (event: Event): void => {
    if (event != undefined) {
      event.preventDefault();
    }

    setCurrentIndex((prevIndex) => prevIndex - 1);
  };

  const onNext = (event: Event): void => {
    if (event != undefined) {
      event.preventDefault();
    }

    setCurrentIndex((prevIndex) => prevIndex + 1);
  };

  return (
    <SliderItem
      hasNext={currentIndex + 1 < props.images.length}
      hasPrev={currentIndex > 0}
      onNext={onNext}
      onPrev={onPrev}
      onCloseModal={onCloseModal}
      imageSrc={props.images[currentIndex]}
    />
  );
};

const MemoizedSlider = React.memo(Slider);
export { MemoizedSlider as Slider };
