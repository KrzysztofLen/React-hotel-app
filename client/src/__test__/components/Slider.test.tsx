import React from 'react';
import { shallow } from 'enzyme';
import { Slider } from '../../components/Slider/Slider';

describe('[Component] Slider', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Slider images={['1', '2', '3']} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Slider component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return undefined if no images provided', () => {
    const wrapper = shallow(<Slider images={[]} />);
    expect(wrapper.prop('imageSrc')).toBeUndefined();
  });

  it('should return correct image prop with default state', () => {
    expect(wrapper.props().imageSrc).toEqual('2');
  });

  //   it('should increment state onNext', () => {
  //     const onNext = jest.fn();
  //     wrapper.onClick();
  //     expect(setState).toHaveBeenCalledWith(1);
  //   });

  //   it('should decrement state onPrev', () => {
  //     const wrapper = shallow(<Slider images={['1', '2']} />);
  //     wrapper.setState({ currentIndex: 2 });
  //     wrapper.instance().onPrev();
  //     wrapper.update();
  //     expect(wrapper.state('currentIndex')).toBe(1);
  //   });
});
