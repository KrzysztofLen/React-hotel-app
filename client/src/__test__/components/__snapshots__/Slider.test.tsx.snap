// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Slider should render Slider component correctly 1`] = `
<SliderItem
  hasNext={true}
  hasPrev={true}
  imageSrc="2"
  onCloseModal={[Function]}
  onNext={[Function]}
  onPrev={[Function]}
/>
`;
